### PR TITLE
Replaced deprecated preg_replace /e + removed unused function

### PR DESCRIPTION
**Pull request by animaux**
- Replaced deprecated preg_replace /e in lib/class.search_index.php
- fixes https://github.com/symphonists/search_index/issues/56

### CODE REVIEW

**Comment by nitriques** [comment]
Should I wait for #56 tu be pushed and fix as well ?

File: lib/class.search_index.php [hunk 1]
```diff
@@ -207,8 +207,16 @@ public function saveEntryKeywords($entry_id, $data) {
...
+    $data = preg_replace_callback('~&#x([0-9a-f]+);~i', 
+      function($m) { return chr(hexdec($m[1])); },
+      $data
+		);
+		
+	  $data = preg_replace_callback('~&#([0-9]+);~', 
+      function($m) { return chr($m[1]); },
+      $data
+    );
+
```
Comment (nitriques) [inline lib/class.search_index.php]: Could you try to fix the indentation here please ?

**Review by nitriques — CHANGES_REQUESTED** [review]

**Comment by animaux** [comment]
@nitriques yes, will try to fix #56 as well. My editor uses spaces not tabs for indentation. I guess that’s waht you like to be fixed?

**Comment by animaux** [comment]
Indentation should be fixed now.

**Comment by nitriques** [comment]
> . I guess that’s what you like to get fixed?

Yeah. The best thing is to keep with the original indentation of the file. If it's tabs, continue with tabs, if it's spaces, go with spaces. Which text editor do you use ?

**Comment by animaux** [comment]
I’m using BBEdit. In my current settings the indentation looked identical to me. Had to switch on invisibles to notice the difference :) Will do the deletion and update the PR on monday. Thanks Nicolas!

**Comment by nitriques** [comment]
Thanks a lot! You do not need to create a new PR, just push to this one and add Fix #56 in the commit message !

**Comment by nitriques** [comment]
> Had to switch on invisibles to notice the difference

Yeah I always have that on ;) Sadly BBEdit does not seems to support auto-detection like Sublime does.

**Review by nitriques — APPROVED** [review]

**Comment by animaux** [comment]
Fixes https://github.com/symphonists/search_index/issues/56.

**Comment by animaux** [comment]
@nitriques please pull! `:)` or will this be pulled automatically if I close it, since it is approved?

**Comment by nitriques** [comment]
I've merged in integration first, you might have to flush your local master. out in https://github.com/symphonists/search_index/releases/tag/0.9.5 !

**Comment by animaux** [comment]
Great, merci!

**Pull request merged**